### PR TITLE
Implement Lock.swift on Windows

### PR DESCRIPTION
**Pull request by jmittert**

### CODE REVIEW

File: Sources/Basic/Lock.swift [hunk 1]
```diff
@@ -53,6 +57,30 @@ public final class FileLock {
     ///
     /// Note: This method can throw if underlying POSIX methods fail.
     public func lock() throws {
+#if os(Windows)
```
Comment (aciidb0mb3r) [inline Sources/Basic/Lock.swift]: Minor note on indenting:

- `#if` statements should be indented two-space less than the regular code. See example: https://github.com/apple/swift-package-manager/blob/master/Sources/PackageDescription4/Package.swift#L68
- SwiftPM uses 4-space indent for regular code.
- When breaking a method into multiple lines, indent the parameters to the left of the call: Ex: https://github.com/apple/swift-package-manager/blob/master/Sources/Workspace/Workspace.swift#L370

**Comment by jmittert** [comment]
Fixed up the indenting

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test linux

File: Sources/Basic/Lock.swift [hunk 2]
```diff
@@ -36,7 +36,11 @@ enum ProcessLockError: Swift.Error {
 /// by mutiple instances of a process. The `FileLock` is not thread-safe.
 public final class FileLock {
     /// File descriptor to the lock file.
+  #if os(Windows)
+    private var h: HANDLE?
```
Comment (unnamedd) [inline Sources/Basic/Lock.swift]: Why not at least `handle` instead of `h` for this variable?

File: Sources/Basic/Lock.swift [hunk 3]
```diff
@@ -36,7 +36,11 @@ enum ProcessLockError: Swift.Error {
...
+  #else
     private var fd: CInt?
```
Comment (unnamedd) [inline Sources/Basic/Lock.swift]: What `fd` stands for?

Comment (aciidb0mb3r) [reply Sources/Basic/Lock.swift]: file descriptor

File: Sources/Basic/Lock.swift (hunk 2 above)
Comment (aciidb0mb3r) [reply Sources/Basic/Lock.swift]: Renaming to `handle` would be nice to have. Feel free to make a PR :)

Comment (unnamedd) [reply Sources/Basic/Lock.swift]: @aciidb0mb3r PR #2093 ;)

**Pull request merged**